mlflow/alembic/versions/451aebb31d03_add_metric_step.py

```diff
@@ -0,0 +1,35 @@
+"""add metric step
+
+Revision ID: 451aebb31d03
+Revises: 
+Create Date: 2019-04-22 15:29:24.921354
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '451aebb31d03'
+down_revision = None
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column('metrics', sa.Column('step', sa.BigInteger(), nullable=False, server_default='0'))
+    # Use batch mode so that we can run "ALTER TABLE" statements against SQLite
+    # databases (see more info at https://alembic.sqlalchemy.org/en/latest/
+    # batch.html#running-batch-migrations-for-sqlite-and-other-databases)
+    with op.batch_alter_table("metrics") as batch_op:
+        batch_op.drop_constraint(constraint_name='metric_pk')
```

Comment:
This fails against MySQL:

```
NotImplementedError: No generic 'DROP CONSTRAINT' in MySQL - please specify constraint type
```

Based on the docs, it looks like the `type_` attribute needs to be specified for MySQL databases (super convenient edge case!). `batch_op.drop_constraint(constraint_name='metric_pk', type_='primary')` seems to work! (Reference: https://alembic.sqlalchemy.org/en/latest/ops.html#alembic.operations.Operations.drop_constraint).

Reply:
Ah, great catch - thanks for finding the fix!